dev_guide/builds.adoc

```diff
@@ -1025,10 +1025,112 @@ is *dockerhub*:
 == Using Private Repositories for Builds
 
 Supply valid credentials to build an application from a private repository.
-Currently, only SSH key based authentication is supported. The repository keys
-are located in the `$HOME/.ssh/` directory, and are named `id_dsa.pub`,
-`id_ecdsa.pub`, `id_ed25519.pub`, or `id_rsa.pub` by default. Generate SSH key
-credentials with the following command:
+
+Currently two types of authentication are supported. Basic username-password
+and SSH key based authentication.
+
+[#basic-authentication]
+=== Basic Authentication
+
+Basic authentication requires either combination of `username` and `password`,
+or a `token` to authenticate against the SCM server. A `CA certificate` file,
+or a `.gitconfig` file can be attached.
+
+A link:dev_guide/secrets[`*secret*`] is used to store your keys.
+
+. Create the `*secret*` first before using the SSH key to access the private
+repository:
++
+====
+----
+$ oc secrets new-basicauth basicsecret --username=USERNAME --password=PASSWORD
+----
+====
+
+.. To create a Basic Authentication Secret with a token:
++
+====
+----
+$ oc secrets new-basicauth basicsecret --password=TOKEN
+----
+====
+
+.. To create a Basic Authentication Secret with a CA certificate file:
++
+====
+----
+$ oc secrets new-basicauth basicsecret --username=USERNAME --password=PASSWORD --ca-cert=FILENAME
+----
+====
+
+.. To create a Basic Authentication Secret with a `.gitconfig` file:
++
+====
+----
+$ oc secrets new-basicauth basicsecret --username=USERNAME --password=PASSWORD --gitconfig=FILENAME
+----
+====
+
+. Add the `*secret*` to the builder service account:
++
+====
+----
+$ oc secrets add serviceaccount/builder secrets/basicsecret
+----
+====
+
+. Add a `*sourceSecret*` field into the `*source*` section inside the
+`*BuildConfig*` and set it to the name of the `*secret*` that you created, in
+this case `*basicsecret*`:
++
+====
+
+----
+{
+  "apiVersion": "v1",
+  "kind": "BuildConfig",
+  "metadata": {
+    "name": "sample-build",
+  },
+  "parameters": {
+    "output": {
+      "to": {
+        "name": "sample-image"
+      }
+    },
+    "source": {
+      "git": {
+        "uri": "https://github.com/user/app.git" <1>
+      },
+      "sourceSecret": {
+        "name": "basicsecret"
+      },
+      "type": "Git"
+    },
+    "strategy": {
+      "sourceStrategy": {
+        "from": {
+          "kind": "ImageStreamTag",
+          "name": "python-33-centos7:latest"
+        }
+      },
+      "type": "Source"
+    }
+  }
+----
+<1> The URL of private repository, accessed by a a basic authentication, is usually
+in the form `https://github.com/user/app.git<username>/<repository>.git`.
```

Comment:
Just saw the example below for the sshauth... I would say that you can mention that the URL must be https or http for basic auth to work.